### PR TITLE
[PL-24913]: Handle the error raised while creating a multipart input

### PR DESCRIPTION
**Pull request by DeepakPatankar**

### CODE REVIEW

File: scm/multipart.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+func (mw *MultipartWriter) Close() {
+	mw.Writer.Close()
+}
+
+func (mw *MultipartWriter) FormDataContentType() string {
+	return mw.Writer.FormDataContentType()
+}
+
+type MultipartWriter struct {
```
Comment (allofthesepeople) [inline scm/multipart.go:23]: nit: typically the struct definition comes before the methods.

File: scm/driver/stash/stash.go [hunk 2]
```diff
@@ -79,26 +79,29 @@ func (c *wrapper) do(ctx context.Context, method, path string, in, out interface
 		case *contentCreateUpdate:
 			// add the content to the multipart
 			var b bytes.Buffer
-			w := multipart.NewWriter(&b)
+			mw := &scm.MultipartWriter{Writer: multipart.NewWriter(&b)}
 			// add the other fields
-			if content.Message != "" {
-				_ = w.WriteField("content", string(content.Content))
+			if string(content.Content) != "" {
```
Comment (allofthesepeople) [inline scm/driver/stash/stash.go:84]: the empty string checks are handled within the `mw.Write` method, so could probably lose these additional checks. 

Though see comment below on whether the general purpose struct should be checking for empty values

File: scm/multipart.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package scm
+
+import "mime/multipart"
+
+func (mw *MultipartWriter) Write(f, v string) {
+	if mw.Error == nil {
+		return
+	}
+	if v == "" {
```
Comment (allofthesepeople) [inline scm/multipart.go:9]: Should possibly leave a note in the method comment about empty strings not being written as this differs from the mime/multipart behaviour it's wrapping.

**Review by allofthesepeople — APPROVED** [review]
couple of nits but nice simplification.

File: scm/driver/stash/stash.go (hunk 2 above)
Comment (DeepakPatankar) [reply scm/driver/stash/stash.go]: done

File: scm/multipart.go (hunk 3 above)
Comment (DeepakPatankar) [reply scm/multipart.go:14]: done

File: scm/multipart.go (hunk 1 above)
Comment (DeepakPatankar) [reply scm/multipart.go]: done

File: scm/multipart.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package scm
```
Comment (tphoney) [inline scm/multipart.go:1]: So this functionality is only used by stash/content. I would move this helper method in under stash 
or 
into scm/driver/stash/stash.go

**Review by tphoney — CHANGES_REQUESTED** [review]

File: scm/multipart.go (hunk 4 above)
Comment (tphoney) [reply scm/multipart.go:1]:
```
➜  go-scm git:(no_project) grep -irsn Multipart .
./scm/driver/bitbucket/bitbucket.go:13: "mime/multipart"
./scm/driver/bitbucket/bitbucket.go:74:                 // add the content to the multipart
./scm/driver/bitbucket/bitbucket.go:77:                 w := multipart.NewWriter(&b)
./scm/driver/bitbucket/bitbucket.go:95:                 // write the multipart response to the body
./scm/driver/bitbucket/bitbucket.go:97:                 // write the content type that contains the length of the multipart
./scm/driver/bitbucket/bitbucket.go:103:                        writer := multipart.NewWriter(body)
```
nothing else is using it yet. I would definitely make it only for stash for now. Until another git provider uses it

**Review by tphoney — APPROVED** [review]

**Pull request merged**